src/types/register.ts

```diff
@@ -11,7 +12,7 @@ export interface Register {
 
 export type RegisteredRouter = Register extends { router: infer TRouter }
   ? TRouter
-  : Router
+  : Router<any>
```

Comment:
Curious why this any is needed

Reply:
internally we use `useRouter()` in RouterLink which will never have Registered router and `Router` without the `<any>` has type never for `router.push` source.

<img width="617" alt="image" src="https://github.com/kitbagjs/router/assets/6098901/9a8f6690-506a-4e78-9128-ec0136e70009">

my thinking is, if a developer doesn't provide register interface, the router they get back from `useRouter` should probably be pretty wide?